docusaurus/docs/how-to-guides/data-source-plugins/data-source-configuration.md

```diff
@@ -0,0 +1,137 @@
+**Example:** [Datadog](https://www.datadoghq.com/about/resources/)
+
+## Description and required fields
+
+We recommend using the `DataSourceDescription` component from the [@grafana/plugin-ui](https://github.com/grafana/plugin-ui) repository. The plugin name and documentation link are required, but details about other fields is optional. The plugin name and documentation link are required, but details about other fields is optional.
+
+For required fields, avoid manually typing `*`. Instead, pass the `required` prop to the `InlineField` component. This prop automatically adds a `*` to the label.
+
+**Example of DataSourceDescription component:**
+
+![Description](/img/data-source-config/config-data-source-description.png)
+
+**Example of required field:**
+
+![Connection](/img/data-source-config/config-connection.png)
+
+## Sections
+
+Divide the data source configuration page into multiple sections using the `ConfigSection` component.
+
+- The `section` description property is optional but recommended if the section contains more than one field.
+- For sections with optional or rarely used settings, it's best to keep them collapsed by default.
+- Use the `ConfigSubSection` component for organizing subsections within a section. Avoid separating subsections with horizontal lines.
+
+**Example of sections:**
+
+![Sections](/img/data-source-config/config-sections.png)
+
+## Authentication component
+
+Use the `Authentication` component for handling authentication on the configuration page. This component is already wrapped inside a `ConfigSection`, so additional wrapping is unnecessary.
+
+If your data source uses the deprecated `DataSourceHttpSettings` component for authentication, replace it with the `Authentication` component. For guidance on migrating from `DataSourceHttpSettings`, see the section below.
+
+**Example of authentication component:**
+
+![Authentication](/img/data-source-config/config-authentication.png)
+
+## DataSourceHttpSettings migration
+
+The `DataSourceHttpSettings` component is deprecated, and you should migrate to newer components. We’ve introduced three components to streamline this process:
+
+- **Connection section**
+- **Authentication** (refer to [Authentication section](#authentication-component))
+- **Advanced settings subsection**
+
+To migrate from `DataSourceHttpSettings` to the new 'Auth' component, refer to our [documentation](https://github.com/grafana/plugin-ui/blob/main/src/components/ConfigEditor/Auth/README.md).
+
+### Edge case handling
+
+If your authentication method doesn't require fields, include text explaining this. Add descriptions or external links as needed.
+
+**Example of edge case:**
+
+![Edge cases](/img/data-source-config/config-authentication-component.png)
+
+### Additional settings
+
+Additional settings are optional and provide users with extra control over the data source plugin. These settings should:
+
+- Be placed in a separate section using the `ConfigSection` component.
```

Comment:
As we are going to keep (for now) the documentation asking to use `DataSourceHttpSettings` let's remove references to the `ConfigSection` component